Services/Init/classes/class.ilInitialisation.php

```diff
@@ -1407,6 +1406,8 @@ public static function resumeUserSession()
             !$GLOBALS['DIC']['ilAuthSession']->isAuthenticated() or
             $GLOBALS['DIC']['ilAuthSession']->isExpired()
         ) {
+            ilSession::_destroy($_COOKIE[session_name()], ilSession::SESSION_CLOSE_EXPIRE);
```

Comment:
@amichelsdatabay Do we need a `if (isset($_COOKIE[session_name()]))` check here?

Reply:
I added 
```
if ($GLOBALS['DIC']['ilAuthSession']->isExpired()) {
```

The cookie is always set if the session is expired because to expire the session it has to be looked up before in the database